### PR TITLE
refactor: make `enum RdbEngine` test-only

### PR DESCRIPTION
**Pull request by laysakura**
## Motivation and background

- `enum RdbEngine` has led to low cohesion codes with conditional branches
  - Then, I wanted to remove `enum RdbEngine`
- Luckily, I succeeded in removing all the dependencies to the `RdbEngine` in main codes but some test codes are making full use of it
  - So I decieded to leave `enum RdbEngine` in test package

## Major changes

- move `enum RdbEngine` from main to test
  - all main classes no longer have `RdbEngine` fields, parameters, and result types
- remove `JdbcConfig.getRdbEngine()`

### CODE REVIEW

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

File: core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java [hunk 1]
```diff
@@ -77,4 +79,8 @@ void dropIndexExecute(Connection connection, String schema, String table, String
   default String encloseFullTableName(String schema, String table) {
     return enclose(schema) + "." + enclose(table);
   }
+
+  SelectQuery buildSelectQuery(SelectQuery.Builder builder, int limit);
+
+  UpsertQuery buildUpsertQuery(UpsertQuery.Builder builder);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java:85]: Not sure these methods are proper abstractions. But, I have no specific ideas for now, so let's keep them. And we can revisit it in the future if needed. Thanks.

**Review by brfrn169 — APPROVED** [review]
Left one comment, but LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**